src/components/profileemailinfo.main.jsx

```diff
@@ -0,0 +1,180 @@
+/**
+ * Copyright © 2018 Elastic Path Software Inc. All rights reserved.
+ *
+ * This is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This software is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this license. If not, see
+ *
+ *     https://www.gnu.org/licenses/
+ *
+ *
+ */
+
+import React from 'react';
+import PropTypes from 'prop-types';
+import intl from 'react-intl-universal';
+import { login } from '../utils/AuthService';
+import { cortexFetch } from '../utils/Cortex';
+import './profileemailinfo.main.less';
+
+const Config = require('Config');
+
+class ProfileemailinfoMain extends React.Component {
+  static propTypes = {
+    profileInfo: PropTypes.objectOf(PropTypes.any).isRequired,
+    onChange: PropTypes.func.isRequired,
+  }
+
+  constructor(props) {
+    super(props);
+    const { profileInfo } = this.props;
+    const email = profileInfo._emails[0]._element ? profileInfo._emails[0]._element[0].email : '';
```

Comment:
`profileInfo ` can be undefined at this point. Need to do a proper check of `profileInfo` as well.